cluster/gce/gci/configure-helper-common.sh

```diff
@@ -1424,7 +1424,112 @@ scrape_configs:
     static_configs:
     - targets: ['127.0.0.1:2379','127.0.0.1:8080','127.0.0.1:10251','127.0.0.1:10252']
 EOF
-  ./prometheus --config.file="/tmp/prometheus-metrics.yaml" --web.listen-address=":9090" --web.enable-admin-api &
+  nohup ./prometheus --config.file="/tmp/prometheus-metrics.yaml" --web.listen-address=":9090" --web.enable-admin-api > prometheus.log 2>&1 &
+
+}
+
+
+function collecting-pprof {
+  local ports=$1
+  local name=$2
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/profile -o ${name}-profile-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/profile -o ${name}-profile-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/heap -o ${name}-heap-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/heap -o ${name}-heap-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/goroutine -o ${name}-goroutine-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/goroutine -o ${name}-goroutine-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/mutex -o ${name}-mutex-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/mutex -o ${name}-mutex-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/block -o ${name}-block-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/block -o ${name}-block-${CURRENTTIME}.pprof
+  echo "curl http://127.0.0.1:${ports}/debug/pprof/threadcreate -o ${name}-threadcreate-${CURRENTTIME}.pprof"
+  curl http://127.0.0.1:${ports}/debug/pprof/threadcreate -o ${name}-threadcreate-${CURRENTTIME}.pprof
+}
+
+# Starts collecting profiling files.
+function start-collect-pprof {
+  echo "Start to collect profiling files"
+  mkdir -p /var/log/pprof
+  while true; do
+    cd /var/log/pprof
+    CURRENTTIME=`date +"%Y-%m-%d-%T"`
+    mkdir "${CURRENTTIME}"
+    cd "${CURRENTTIME}"
+
+    COMPONENTS_PORTS="8080"
+    COMPONENTS_NAME="kube-apiserver"
+    echo "Collecting kube-apiserver pprof at ${CURRENTTIME}"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+
+    COMPONENTS_PORTS="2379"
+    COMPONENTS_NAME="etcd"
+    echo "Collecting etcd pprof at ${CURRENTTIME}"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+
+    COMPONENTS_PORTS="10251"
+    COMPONENTS_NAME="kube-scheduler"
+    echo "Collecting kube-scheduler pprof at ${CURRENTTIME}"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+    
+    COMPONENTS_PORTS="10252"
+    COMPONENTS_NAME="kube-controller-manager"
+    echo "Collecting kube-controller-manager pprof"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+
+    COMPONENTS_PORTS="10250"
+    COMPONENTS_NAME="kubelet"
+    echo "Collecting kubelet pprof"
+    collecting-pprof ${COMPONENTS_PORTS} ${COMPONENTS_NAME} 
+    sleep 1800
+  done &
+}
+
+
+# Starts collecting profiling files.
+function start-collect-partitionserver-pprof {
```

Comment:
where's this being called?

Reply:
in cluster/gce/gci/configure-helper.sh. you can find called information in same commit:
```
if [[ "${ENABLE_PPROF_DEBUG:-false}" == "true" ]]; then
  start-collect-pprof &  #### start collect profiling files
  fi
```

Reply:
this code piece is calling start-collect-pprof while the function name here is start-collect-partitionserver-pprof

Reply:
you are correct, just fixed and code in cluster/gce/gci/partitionserver-configure-helper.sh
```
if [[ "${ENABLE_PPROF_DEBUG:-false}" == "true" ]]; then
    start-collect-partitionserver-pprof &  #### start collect profiling files
  fi
```

Reply:
thanks. considered this resolved.